lib/promscrape/scrapework.go

```diff
@@ -905,10 +907,28 @@ func (sw *scrapeWork) addRowToTimeseries(wc *writeRequestCtx, r *parser.Row, tim
 		Value:     r.Value,
 		Timestamp: sampleTimestamp,
 	})
+	// Add Exemplars to Timeseries
+	exemplarsLen := len(wc.exemplars)
+	if len(r.Exemplar.Tags) > 0 {
+		exemplarLabels := []prompbmarshal.Label{}
```

Comment:
It's better to pre-allocate the required capacity of slice by using `make([]prombmarshal.Label{}, 0, len(r.Exemplar.Tags))`

Reply:
yes I agree however I think an even slightly more efficient approach would be:
make([]prompbmarshal.Label{}, len(r.Exemplar.Tags)) and use direct assignment.